### PR TITLE
improve image tests

### PR DESCRIPTION
**Pull request by dorschw**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/46177

## Description
- Added a retry mechanism to the validation that checks remote images.
- Only checking for http status code without downloading the image checked (thanks @ilaner for the idea)

## Must have
- [x] Tests

### CODE REVIEW

**Review by DeanArbel — APPROVED** [review]
Impressive changes. Well done.

File: demisto_sdk/commands/common/errors.py [hunk 1]
```diff
@@ -1646,14 +1646,19 @@ def invalid_readme_insert_image_link_error(path):
...
-            'pack_readme_relative_error': Errors.pack_readme_image_relative_path_error,
-            'general_readme_relative_error': Errors.invalid_readme_image_relative_path_error,
-            'general_readme_absolute_error': Errors.invalid_readme_image_absolute_path_error,
-            'branch_name_readme_absolute_error': Errors.branch_name_in_readme_image_absolute_path_error,
-            'insert_image_link_error': Errors.invalid_readme_insert_image_link_error
-        }.get(error_type, lambda x: f'Something went wrong when testing {x}')(path)
+    def invalid_readme_image_error(path: str, error_type: str, http_code: Optional[int] = None):
+        error = 'Error in readme image: '
+        if http_code:
+            error += f'got HTTP response code {http_code}'
```
Comment (tkatzir) [inline demisto_sdk/commands/common/errors.py:1652]: Need to add an extra space at the end?

File: demisto_sdk/commands/common/errors.py [hunk 2]
```diff
@@ -1646,14 +1646,19 @@ def invalid_readme_insert_image_link_error(path):
...
     @error_code_decorator
-    def invalid_readme_image_error(path, error_type):
-        return 'Error in readme image:\n' + {
-            'pack_readme_relative_error': Errors.pack_readme_image_relative_path_error,
-            'general_readme_relative_error': Errors.invalid_readme_image_relative_path_error,
-            'general_readme_absolute_error': Errors.invalid_readme_image_absolute_path_error,
-            'branch_name_readme_absolute_error': Errors.branch_name_in_readme_image_absolute_path_error,
-            'insert_image_link_error': Errors.invalid_readme_insert_image_link_error
-        }.get(error_type, lambda x: f'Something went wrong when testing {x}')(path)
+    def invalid_readme_image_error(path: str, error_type: str, http_code: Optional[int] = None):
```
Comment (tkatzir) [inline demisto_sdk/commands/common/errors.py:1649]: Accept the `reason` param as well

File: demisto_sdk/commands/common/errors.py (hunk 1 above)
Comment (tkatzir) [reply demisto_sdk/commands/common/errors.py:1652]: Print the `reason` param as well

**Review by tkatzir — CHANGES_REQUESTED** [review]
Great work!
Missing an extra space at the end, for readability.

**Comment by idovandijk** [comment]
@dorschw can u please explain to me in short what this change is?

**Comment by dorschw** [comment]
> @dorschw can u please explain to me in short what this change is?

Updated PR description:
```
## Description
- Added a retry mechanism to the validation that checks remote images.
- Only checking for http status code without downloading the image checked (thanks @ilaner for the idea)
```

**Review by idovandijk — APPROVED** [review]
So nothing for me to review here, if they approved it's fine

**Review by tkatzir — APPROVED** [review]

**Pull request merged**